src/main/java/com/avispl/symphony/dal/communicator/barco/e2gen2/BarcoE2Communicator.java

```diff
@@ -0,0 +1,998 @@
+	 * @param numberOfSource Number of sources that are assigned to this destination
+	 */
+	private void populateRouting(Map<String, String> stats, String methodName, String screenDestName, String currentSourceName,
+			List<String> sourceList, List<AdvancedControllableProperty> controls, int numberOfSource) {
+		if (Objects.equals(currentSourceName, BarcoE2Constant.DOUBLE_QUOTES)) {
+			currentSourceName = BarcoE2Constant.NONE;
+		}
+		stats.put(String.format(BarcoE2Constant.GROUP_HASH_TAG_MEMBER, methodName, screenDestName), currentSourceName);
+		controls.add(createDropdown(String.format(BarcoE2Constant.GROUP_HASH_TAG_MEMBER, methodName, screenDestName), currentSourceName, sourceList));
+		if (numberOfSource > 1) {
+			stats.put(String.format("%s#%s%s", methodName, screenDestName, BarcoE2Constant.DESTINATION_STATUS), BarcoE2Constant.DESTINATION_MIXED);
+		}
+	}
+
+	/**
+	 * Routing control:  Assign a new source to Aux Destination
+	 *
+	 * @param auxName Name of Aux Destination
+	 * @param sourceName The source that will be assigned to the Aux Destination
+	 * @return boolean true/false based on the response.
+	 */
+	private boolean changeAuxContent(String auxName, String sourceName) {
+		try {
+			Map<Integer, String> listSources = getListSource();
+			int newSourceIndex = 0;
+			Map<Object, Object> params = new HashMap<>();
+			Map<Object, Object> changeAuxParams = new HashMap<>();
+			for (Entry<Integer, String> entry : listSources.entrySet()) {
+				if (entry.getValue().equals(sourceName)) {
+					newSourceIndex = entry.getKey();
+					break;
+				}
+			}
+			params.put(BarcoE2Constant.TYPE, BarcoE2Constant.DESTINATION_AUX_TYPE);
+			JsonNode response = getByMethod(BarcoE2Constant.METHOD_LIST_DESTINATIONS, params).get(BarcoE2ControllingMetric.AUX_DESTINATION.getName());
+			int currentAuxDestId = BarcoE2Constant.NOT_MATCH_AUX_ID;
+			for (int i = 0; i < response.size(); i++) {
+				JsonNode auxResponse = response.get(i);
+				if (auxResponse != null && Objects.equals(auxName, auxResponse.get(BarcoE2Constant.NAME).asText())) {
+					currentAuxDestId = response.get(i).get(BarcoE2Constant.ID).asInt();
+				}
+			}
+			changeAuxParams.put(BarcoE2Constant.ID, currentAuxDestId);
+			changeAuxParams.put(BarcoE2Constant.PGM_LAST_SRC_INDEX, newSourceIndex);
+			JsonNode changeAuxContentResult = getByMethod(BarcoE2Constant.METHOD_CHANGE_AUX_CONTENT, changeAuxParams);
+			if (changeAuxContentResult != null && changeAuxContentResult.get(BarcoE2Constant.SUCCESS_STATUS) != null) {
+				return changeAuxContentResult.get(BarcoE2Constant.SUCCESS_STATUS).asInt() == 0;
+			}
+			throw new CommandFailureException(this.getAddress(), "changeAuxContent", "Failed to call changeAuxContent");
+		} catch (Exception e) {
+			logger.error("Failed to assign source to super/aux destination");
+			throw new CommandFailureException(this.getAddress(), "changeAuxContent or changeSuperAuxContent", "Failed to assign source to this destination");
```

Comment:
is there a need to log an error, if it's getting thrown afterwards?

Reply:
I deleted the logger here and add the 2nd parameter.